tools/lsp/preview.rs

```diff
@@ -83,6 +88,140 @@ fn set_design_mode(enable: bool) {
     configure_design_mode(enable);
 }
 
+fn follow_child(element: &ElementRc) -> ElementRc {
+    let elem = element.borrow();
+    if elem.repeated.is_some() {
+        if let i_slint_compiler::langtype::ElementType::Component(base) = &elem.base_type {
+            return base.root_element.clone();
+        }
+    }
+    element.clone()
+}
+
+fn lsp_element_position(element: &ElementRc) -> (String, usize, usize, usize, usize) {
```

Comment:
I wonder if we wouldn't benefit to have an actual type instead of `(String, usize, usize, usize, usize)`   Maybe something like SourceRange or something.